lightning_pose/utils/pca.py

```diff
@@ -372,71 +304,6 @@ def __call__(self) -> int:
             )  # find that integer that crosses the minimum explained variance
 
 
-# TODO: the function below was in usage until Mar 21, 2022. Integrating it into the pca class
-@typechecked
-def compute_pca_reprojection_error(
-    clean_pca_arr: TensorType["samples", "observation_dim"],
-    kept_eigenvectors: TensorType["latent_dim", "observation_dim"],
-    mean: TensorType["observation_dim"],
-) -> TensorType["samples", "observation_dim_div_by_two"]:
-
-    # first verify that the pca array has observations divisible by 2
-    # (corresponding to (x,y) coords)
-    assert clean_pca_arr.shape[1] % 2 == 0
-
-    # mean-center
-    clean_pca_arr = clean_pca_arr - mean.unsqueeze(0)
-
-    # project down into low-d space, then back up to observation space
-    reprojection_arr = (
-        clean_pca_arr @ kept_eigenvectors.T @ kept_eigenvectors
-    )  # e.g., (214, 4) X (4, 3) X (3, 4) = (214, 4) as we started
-
-    # compute difference between original array and its reprojection
-    # multiview:
-    # - num_samples: num_samples X num_bodyparts
-    # - observation_dim: 2 x num_views
-    # singleview:
-    # - num_samples: num_samples
-    # - observation_dim: 2 X num_bodyparts
-    diff = clean_pca_arr - reprojection_arr
-
-    # reshape to get a per sample/keypoint values:
-    diff_arr_per_keypoint = diff.reshape(diff.shape[0], diff.shape[1] // 2, 2)
-    reprojection_loss = torch.linalg.norm(diff_arr_per_keypoint, dim=2)
-
-    # print("reprojection_loss.shape: {}".format(reprojection_loss.shape))
-    # print("diff: {}".format(diff[:5, :]))
-    # print("diff_arr_per_keypoint: {}".format(diff_arr_per_keypoint[:5, :, :]))
-
-    return reprojection_loss
-
-
-@typechecked
-def add_params_to_loss_dict(
```

Comment:
it's been a while since I looked at this guy. what's going on with it? do we still use it?